odl/tomo/backends/astra_setup.py

```diff
@@ -0,0 +1,581 @@
+    # this was possible only for 2D geometries. Thus, standard ASTRA
+    # projection geometries do not have to be rescaled any more by the (
+    # isotropic) voxel size.
+    if isinstance(geometry, Parallel2dGeometry):
+        det_width = geometry.det_grid.stride[0]
+        det_count = geometry.det_grid.shape[0]
+        angles = geometry.motion_grid.coord_vectors[0]
+        proj_geom = astra.create_proj_geom(
+            'parallel', det_width, det_count, angles)
+
+    elif isinstance(geometry, FanFlatGeometry):
+        det_width = geometry.det_grid.stride[0]
+        det_count = geometry.det_grid.shape[0]
+        angles = geometry.motion_grid.coord_vectors[0]
+        source_origin = geometry.src_radius
+        # False label in PyASTRA doc as `source_det`
+        origin_det = geometry.det_radius
+        proj_geom = astra.create_proj_geom(
+            'fanflat', det_width, det_count, angles, source_origin, origin_det)
+
+    elif isinstance(geometry, Parallel3dGeometry):
+        det_width = geometry.det_grid.stride[0]
+        det_height = geometry.det_grid.stride[1]
+        det_row_count = geometry.det_grid.shape[1]
+        det_col_count = geometry.det_grid.shape[0]
+        angles = geometry.motion_grid.coord_vectors[0]
+        proj_geom = astra.create_proj_geom(
+            'parallel3d', det_width, det_height, det_row_count,
+            det_col_count, angles)
+
+    elif isinstance(geometry, CircularConeFlatGeometry):
+        det_width = geometry.det_grid.stride[0]
+        det_height = geometry.det_grid.stride[1]
+        det_row_count = geometry.det_grid.shape[1]
+        det_col_count = geometry.det_grid.shape[0]
+        angles = geometry.motion_grid.coord_vectors[0]
+        source_origin = geometry.src_radius
+        # ! In PyASTRA doc falsely labelled `source_det`
+        origin_det = geometry.det_radius
+        proj_geom = astra.create_proj_geom(
+            'cone', det_width, det_height, det_row_count, det_col_count,
+            angles, source_origin, origin_det)
+
+    elif isinstance(geometry, HelicalConeFlatGeometry):
+        det_row_count = geometry.det_grid.shape[1]
+        det_col_count = geometry.det_grid.shape[0]
+        vec = astra_geom_to_vec(geometry)
+        proj_geom = astra.create_proj_geom(
+            'cone_vec', det_row_count, det_col_count, vec)
+    else:
+        raise NotImplementedError('unkown ASTRA geometry type {}.'.format(type(
+            geometry)))
+
+    return proj_geom
+
+
+def astra_data(astra_geom, datatype, data=None, ndim=2):
+    """Create an ASTRA data structure.
+
+    Parameters
+    ----------
+    astra_geom : `dict`
+        ASTRA geometry object for the data creator, must correspond to the
+        given data type
+    datatype : {'volume', 'projection'}
+        Type of the data container
+    data : `DiscreteLpVector` or `None`, optional
+        Data for the initialization of the data structure
```

Comment:
We write only "`DiscreteLpVector`, optional" and explain what happens in the default case. Don't mention `None` as a valid type (at least that's our style so far).